src/ray/raylet/node_manager.cc

```diff
@@ -535,6 +535,11 @@ void NodeManager::FillResourceReport(rpc::ResourcesData &resources_data) {
   resources_data.set_node_manager_address(initial_config_.node_manager_address);
   // Update local cache from gcs remote cache, this is needed when gcs restart.
   // We should always keep the cache view consistent.
+  bool at_capacity = object_manager_.PullManagerAtCapacity();
+  cluster_resource_scheduler_->UpdateObjectPullsQueuedLocally(at_capacity);
```

Comment:
It's a bit odd to poke the message into the cluster resource scheduler. How about either pulling this via a callback, or directly updating the proto below after FillResourceUsage()?